File: examples/javascript/test/elements/fileUpload.spec.js
const { suite } = require('selenium-webdriver/testing');
const {Browser, By, until} = require("selenium-webdriver");
const path = require("path");
const assert = require('node:assert');

suite(function(env) {
  describe('File Upload Test', function() {
    let driver;

    before(async function() {
      driver = await env.builder().build();
    });

    after(() => driver.quit());

    it('Should be able to upload a file successfully', async function() {
      const image = path.resolve('./test/resources/selenium-snapshot.png')

      await driver.manage().setTimeouts({implicit: 5000});

      // Navigate to URL
      await driver.get('https://the-internet.herokuapp.com/upload');
      // Upload snapshot
      await driver.findElement(By.id("file-upload")).sendKeys(image);
      await driver.findElement(By.id("file-submit")).submit();
      
      const revealed = await driver.findElement(By.id('uploaded-files'))
      await driver.wait(until.elementIsVisible(revealed), 2000);
      const data = await driver.findElement(By.css('h3'));
      
      assert.equal(await data.getText(), `File Uploaded!`);
    });

  });
}, { browsers: [Browser.CHROME, Browser.FIREFOX]});

File: examples/ruby/spec/elements/file_upload_spec.rb
# frozen_string_literal: true

require 'spec_helper'

RSpec.describe 'File Upload' do
  let(:driver) { start_session }

  it 'uploads' do
    driver.get('https://the-internet.herokuapp.com/upload')
    upload_file = File.expand_path('../spec_support/selenium-snapshot.png', __dir__)

    file_input = driver.find_element(css: 'input[type=file]')
    file_input.send_keys(upload_file)
    driver.find_element(id: 'file-submit').click

    file_name = driver.find_element(id: 'uploaded-files')
    expect(file_name.text).to eq 'selenium-snapshot.png'
  end
end
